### ISSUE TITLE
Creating a PR to enable file deletions in a commit

### ISSUE THREAD

**Issue by Akrabut** [mheap/octokit-commit-multiple-files#6]
Hello!

I've incorporated this library in my (private) project (great job by the way!).
I need an additional functionality that will allow me to use a single commit for not only file updates and creations, but also file deletions.

Would you want me to create a PR with that?
If so, any guidelines or suggestions?

Thanks

**Comment by mheap** [comment]
That would be excellent! I can see a couple of possibilities

It could be a `delete` type:

```
{
  "message": "This is a submodule commit",
  "files": {
    "my_submodule": {
      "type": "delete"
    }
  }
}
```

Or it could be a new field:

```
    {
      message: "This is a separate commit",
      files: {
        "second.md": "Where should we go today?",
      },
      deleted_files: ["example.md", "another.md"]

    },
```

What do you think?

**Comment by Akrabut** [comment]
The second option looks much more readable, I'll play around with it and open a PR as soon as I'm done.

Any guidelines for tests?

**Comment by mheap** [comment]
Tests please! You can see examples in https://github.com/mheap/octokit-commit-multiple-files/blob/master/create-or-update-files.test.js

We use `nock` to mock the GitHub API

**Comment by Akrabut** [comment]
From my experiments I've found that if our commit will attempt to delete a file that doesn't exist (say it had already been deleted), the entire commit will fail.

This is, in my opinion, pretty bad from a developer experience point of view, given that with current behavior files that don't exist will be created, and those that exist will be updated.

I thought about simply wrapping the `createTree` call in a `try-catch` block that will attempt to create the tree with the deletions and if it fails, discard the deletions and submit commit them.

What do you think?

**Comment by mheap** [comment]
This could go either way:

If the file doesn't exist, then the end result is the same - the file with that name no longer exists in the repo.

However, if the user passes the wrong filename, how will they know that they made a mistake?

How about making it configurable?

```
    {
      message: "This is a separate commit",
      files: {
        "second.md": "Where should we go today?",
      },
      deleted_files: ["example.md", "another.md"],
      ignore_missing: true // Default to false

    },
```

**Comment by Akrabut** [comment]
Could it be that many of the tests fail without changes?
After having fixed `TypeError: Octokit is not a constructor` with:
`const Octokit = require("@octokit/rest");` -> `const Octokit = require("@octokit/rest");`, the result of a test run is:
```
Test Suites: 1 failed, 1 total
Tests:       9 failed, 7 passed, 16 total
```

**Comment by mheap** [comment]
I see that you resolved the issue with the Octokit import 🙌 

What would your recommendation be if one of the `deleted_files` is missing?

**Comment by Akrabut** [comment]
Oh yea, sorry, forgot to handle it.

In case its missing we should just skip the iterations.

Then there's only the case where the `change.retryOnDeleteFailure` is true and `change.filesToDelete` is missing which I believe it could be ignored since it makes no sense that anyone would do that anyway.

**Comment by reiniertimmer** [comment]
Nice work, I just started using this plugin (since I have to do some automated updates to repositories) and like to have this in a single commit. This seems just like the thing for it.

Supporting file delete would be a great addition

**Comment by mheap** [comment]
Resolved in #7. I'll close this issue when I tag a release

**Comment by mheap** [comment]
Version `3.1.0` released to NPM https://www.npmjs.com/package/octokit-commit-multiple-files
